Marketplace/AzureStack.Marketplace.psm1

```diff
@@ -0,0 +1,194 @@
+﻿# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE.txt in the project root for license information.
+
+#requires -Modules AzureStack.Connect
+
+<#
+    .SYNOPSIS
+    Contains 3 functions.
+    Add-GalleryItem: Uploads a gallery item azpkg to Azure stack marketplace.
+    Get-GalleryItem: Gets a gallery item existed in Azure stack marketplace.
+    Remove-GalleryItem: Removes gallery item from Azure stack marketplace.
+#>
+
+Function Add-GalleryItem{
+
+    [CmdletBinding(DefaultParameterSetName='GalleryItemFromLocal')]
+    Param(
+        [Parameter(Mandatory=$true, ParameterSetName='GalleryItemFromLocal')]
+        [ValidateNotNullorEmpty()]
+        [String] $galleryItemLocalPath,
```

Comment:
If this is the gallery item local path, better use the [ValidateScript({Test-Path $_})] to make sure this is a valid path and the file exists